FNML/ontology.md

```diff
@@ -92,21 +109,81 @@ see, e.g. test case [RMLFNOTC0019](https://github.com/RMLio/rml-fno-test-cases/t
 The mapping challenge [Join on literals](https://github.com/kg-construct/mapping-challenges/pull/29) also influences this spec.
 </p>
 
-#### Subject map
+### fnml:execution
 
-When this triples map **does not specify a subject map**, a blank node should be generated for the subject.
+fnml:execution connects the RDF dataset generating triples map via a [fnml:ExecutionTermMap] with a [=fnml:Execution=].
+It has domain [=fnml:ExecutionTermMap=] and range [=fnml:Execution=].
 
-<div class="note">
-We should probably define a new resource that is a subclass of rr:TriplesMap,
-as the description above is disjoint with rr:TriplesMap, as logical source and subjectMap are not optional in [[R2RML]]
-</div>
+### fnml:output
+
+<dfn class="lint-ignore">fnml:output</dfn> connects the RDF dataset generating triples map via a [fnml:ExecutionTermMap] with an output predicate.
+It has domain [=fnml:ExecutionTermMap=] and range [rdf:Property].
+
+### fnml:function
+
+<dfn class="lint-ignore">fnml:function</dfn> connects the [fnml:Execution] with an FnO [=function description=].
+It has domain [=fnml:Execution=] and range [fno:Function](https://w3id.org/function/ontology#Function).
+
+### fnml:predicate
+
+<dfn class="lint-ignore">fnml:predicate</dfn> connects the [fnml:ParameterTermMap] with a parameter predicate.
+It has domain [=fnml:ParameterTermMap=] and range [rdf:Property].
```

Comment:
Similar to above, we shouldn't restrict the range to `rdf:Property` to more easily connect other approaches in the future.